lib/src/metadata/metadata_matcher.dart

```diff
@@ -17,7 +17,14 @@ abstract class MetadataMatcher {
         return fit;
       }
     }
-    return potentialFits[0];
+    return null;
+  }
+
+  static PhoneMetadata getMatchUsingPatterns(

```

Comment:
This does not seem to be used anywhere anymore

Reply:
`getMatchUsingPatternsStrict` was introduced to avoid duplicated code in `getMatchUsingPatterns`. As I wrote, I tried to localize changes maximally, but I suggest redesigning interfaces to avoid exceptions and non-valid metadata matches.

Reply:
Done.